src/bundle.c

```diff
@@ -723,6 +727,53 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	to_install_files = filter_out_existing_files(to_install_files);
 	grabtime_stop(&times);
 
+	/* Step 1.5: Now that we know the bundle is valid, check if we have enough space */
+	if (!skip_diskspace_check) {
+		char *filepath = NULL;
+		ptr = list_head(to_install_bundles);
+		while (ptr) {
+			struct manifest *subman;
+			subman = ptr->data;
+			ptr = ptr->next;
+
+			struct bundle_result *bundle = NULL;
+			bundle = calloc(sizeof(struct bundle_result), 1);
+			/* point to or copy relevant information over for future use */
+			strncpy(bundle->bundle_name, subman->component, BUNDLE_NAME_MAXLEN - 1);
+			bundle->topsize = subman->contentsize;
+			bundle->includes = subman->includes;
+			bundle_info = list_prepend_data(bundle_info, bundle);
+			bundle_size += calculate_size(subman->component, bundle_info, false);
+		}
+		list_free_list_and_data(bundle_info, free);
+
+		/* default is '/', so if '/', check 'usr', else, use the custom path. */
+		if(strcmp("/", path_prefix) == 0) {
+			fs_free = get_available_space("/usr/");
+		}
+		else {
+			string_or_die(&filepath, "%s/%s", path_prefix, "/usr/");
+			fs_free = get_available_space(filepath);
+		}
+
+		if (fs_free < 0){
+			fprintf(stderr, "Error: Could not determine free space in %s\n", filepath);
+			ret = EBUNDLE_INSTALL;
```

Comment:
I prefer to print the same warning you are printing when there's no space, saying that we couldn't determine if there's enough space for the install, but users can override this using --skip-diskspace-check.

Reply:
Done